### PR TITLE
Write linked list

### PR DESCRIPTION
**Pull request by ziolushka**

### CODE REVIEW

File: list.c [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+}
+/* Inserts value at the end of the list. */
+void StringListAdd(char*** list, char* str) {
+
+    const size_t size = strlen(str) + 1;
+
+    //create new node
+    char** node = (char**)malloc(2 * sizeof(char*));
+    node[data] = (char*)malloc(sizeof(char) * size);
+    memcpy(node[data], str, sizeof(char) * size);
```
Comment (Stepbura) [inline list.c:53]: The potential null pointer is passed into 'memcpy' function. Inspect the first argument.

File: list.c [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+}
+/* Replaces every occurrence of the before, in each of the string lists's strings, with after. */
+void StringListReplaceInStrings(char*** list, char* before, char* after) {
+
+    char** curr = *list;
+
+    while (curr != NULL) {
+        if (!strcmp(curr[data], before)) {
+            char* tmp = (char*)malloc(sizeof(char) * (strlen(after) + 1));
+            strcpy(tmp, after);
```
Comment (Stepbura) [inline list.c:164]: The potential null pointer is passed into 'strcpy' function. Inspect the first argument.

File: list.c [hunk 3]
```diff
@@ -0,0 +1,186 @@
+#include <list.h>
+
+#define data 0
+#define next 1
+
+/*Prints list*/
+void StringListPrint(char** list) {
+
+    char** temp = list;
```
Comment (Stepbura) [inline list.c]: Please first verify input parameter and then assign it.

File: list.c [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    if (list == NULL) {
+        printf("List is empty!");
+        return;
+    }
+    printf("List members:\n");
+    while (temp != NULL) {
+        printf("%s\n", (temp[data]));
+        temp = temp[next];
```
Comment (Stepbura) [inline list.c:18]: Cannot convert from 'char *' to 'char **'

**Review by Stepbura — CHANGES_REQUESTED** [review]
Please take a look my comments

File: list.c (hunk 4 above)
Comment (ziolushka) [reply list.c:18]: Resolved

File: list.c (hunk 3 above)
Comment (ziolushka) [reply list.c]: Resolved

File: list.c (hunk 2 above)
Comment (ziolushka) [reply list.c:164]: Resolved

File: list.c (hunk 1 above)
Comment (ziolushka) [reply list.c:53]: Resolved

**Pull request merged**